modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/CommandProcessor.java

```diff
@@ -508,31 +508,29 @@ private void processKillQueryCommand(SqlKillQueryCommand cmd) {
      *
      * @param cmd Sql analyze command.
      */
-    private void processAnalyzeCommand(SqlAnalyzeCommand cmd) {
+    private void processAnalyzeCommand(SqlAnalyzeCommand cmd) throws IgniteCheckedException {
         ctx.security().authorize(SecurityPermission.CHANGE_STATISTICS);
         IgniteStatisticsManager statMgr = ctx.query().getIndexing().statsManager();
         StatisticsTarget[] targets = cmd.targets().stream()
             .map(t -> (t.schema() == null) ? new StatisticsTarget(cmd.schemaName(), t.obj(), t.columns()) : t)
             .toArray(StatisticsTarget[]::new);
 
-        // TODO: save config after GG-32420
-        statMgr.gatherObjectStatisticsAsync(targets);
+        statMgr.updateStatistics(targets);
     }
 
     /**
      * Process refresh statistics command.
      *
      * @param cmd Refresh statistics command.
      */
-    private void processRefreshStatisticsCommand(SqlRefreshStatitsicsCommand cmd) {
+    private void processRefreshStatisticsCommand(SqlRefreshStatitsicsCommand cmd) throws IgniteCheckedException {
         ctx.security().authorize(SecurityPermission.REFRESH_STATISTICS);
         IgniteStatisticsManager statMgr = ctx.query().getIndexing().statsManager();
         StatisticsTarget[] targets = cmd.targets().stream()
             .map(t -> (t.schema() == null) ? new StatisticsTarget(cmd.schemaName(), t.obj(), t.columns()) : t)
             .toArray(StatisticsTarget[]::new);
 
-        // TODO: load config after GG-32420
-        statMgr.gatherObjectStatisticsAsync(targets);
+        statMgr.updateStatistics(targets);
```

Comment:
Should refresh statistics not by target themselves, but by current configuration for specified targets, i.e. need to load config by target and update by loaded config.